### PR TITLE
wallet-ext: create mnemonic with password

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* under feature flag (until everything is using the encrypted mnemonic)
* updates UI of create mnemonic view
* creates and stores the encrypted mnemonic (with bg service) but also stores it plain to not break the existing functionality
* also removed eslint `jsx-no-bind` rule

### Encrypt wallet enabled (env variable `WALLET_ENCRYPTION_ENABLED='true'`)

Screenshot 2022-10-07 at 21 33 10
Screenshot 2022-10-07 at 21 33 39
Screenshot 2022-10-07 at 21 33 54
Screenshot 2022-10-07 at 21 34 05
Screenshot 2022-10-07 at 21 34 10
Screenshot 2022-10-07 at 21 34 20
Screenshot 2022-10-07 at 21 34 55

https://user-images.githubusercontent.com/10210143/194664185-eb034267-490b-4e15-a2ac-907064019758.mov

### Encrypt wallet disabled

Screenshot 2022-10-07 at 21 50 17
Screenshot 2022-10-07 at 21 50 46

https://user-images.githubusercontent.com/10210143/194664357-132c9050-d234-49e3-9197-3e2a58960ce1.mov

part of #4926

### CODE REVIEW

**Comment by randall-Mysten** [comment]
Some of the strings seem to differ from the Figma mocks. Such as Wallet Successfully Created and Backup Recovery Passphrase, and a typo on Create New Wallet on the Encrypt Wallet disabled screen, and Done for the button label

**Comment by mystie711** [comment]
Can you add a video each for import and create flow so it is easier to compare each screen with Figma?
There seems to be a considerable set of UI deltas.

**Comment by pchrysochoidis** [comment]
> Some of the strings seem to differ from the Figma mocks. Such as Wallet Successfully Created and Backup Recovery Passphrase, and a typo on Create New Wallet on the Encrypt Wallet disabled screen, and Done for the button label

@randall-Mysten This view that you refer is the existing one, I am working on it now and will open another PR

**Comment by pchrysochoidis** [comment]
> Can you add a video each for import and create flow so it is easier to compare each screen with Figma? There seems to be a considerable set of UI deltas.

@mystie711 sure give me some time and I will upload one for create. This PR only changes the `create view` (one page)

File: apps/wallet/src/ui/app/shared/input/password/validation.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            letter.toLowerCase() !== letter.toUpperCase() &&
+            letter === letter.toUpperCase()
+        ) {
+            return true;
+        }
+    }
+    return false;
+}
+
+export const passwordValidation = Yup.string()
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/shared/input/password/validation.ts:18]: We should talk through what we want the actual password requirements to be. We talked about using a password strength checker (cc @joyqvq)

Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/shared/input/password/validation.ts:18]: Yeah Joy mentioned it to me but for now I just did what the design had. I added it to the list [here](https://github.com/MystenLabs/sui/issues/4926)

Comment (mystie711) [reply apps/wallet/src/ui/app/shared/input/password/validation.ts:18]: Yes. Thanks. 
The UI copy is meant to be a placeholder until we are figuring out the requirements.
Do we have them @Jordan-Mysten ?

**Review by mystie711 — APPROVED** [review]

**Pull request merged**